### PR TITLE
feat: add YAML schema for reference data

### LINKED ISSUE #130: feat: Reference data YAML schema as first-class file format
## Summary

Loading reference data currently requires either:

1. **Manual construction** -- 30+ lines of `ref.add_energy()`, `ref.add_frequency()`, etc. (as shown in the tutorial)
2. **`load_reference_from_fchk()`** -- a 130-line function buried inside `objective.py` that parses Gaussian `.fchk` files

There's no unified, user-facing file format for reference data. JSON fixtures exist for tests but aren't documented as a user format.

## Proposed Changes

1. Define a `reference.yaml` schema:

```yaml
molecules:
  - name: ch3f
    xyz: ch3f-optimized.xyz
    hessian: ch3f-hessian.npy
    data:
      - kind: energy
        value: -139.7621
        weight: 1.0
      - kind: frequency
        indices: [0, 1, 2, 3, 4, 5, 6, 7, 8]
        source: ch3f-frequencies.txt
        weight: 10.0
      - kind: eigenmatrix
        source: ch3f-hessian.npy
        diagonal_weight: 1.0
        offdiagonal_weight: 0.1
```

2. Implement `ReferenceData.from_yaml()` and `ReferenceData.to_yaml()`
3. `load_reference_from_fchk()` becomes a converter: `.fchk` -> `.yaml`
4. Update tutorial to use YAML loading instead of manual construction
5. Support toggling data types on/off by commenting YAML lines

## Benefits

- Users stop writing 30-line Python scripts to load reference data
- Reproducible training sets are a YAML file in version control
- Easy to share reference datasets between projects

## Files Involved

- `q2mm/optimizers/objective.py` -- add `from_yaml()`/`to_yaml()`, move `load_reference_from_fchk()` to parsers
- `q2mm/parsers/` -- new `reference_yaml.py` or extend existing
- `docs/tutorial.md` -- update to show YAML workflow
- `docs/reference-data.md` -- document the schema

### PR DESCRIPTION
**Pull request by ericchansen**
## Summary

Add YAML schema for reference data, enabling human-readable configuration of QM reference values.

### Changes

- **q2mm/parsers/reference_yaml.py**: New parser (`load_reference_yaml`) and writer (`save_reference_yaml`) for YAML reference data format
- **q2mm/optimizers/objective.py**: Added `from_yaml()` / `to_yaml()` convenience methods on `ReferenceData`
- **test/test_reference_yaml.py**: 32 comprehensive tests covering parsing, writing, round-trip, validation, edge cases
- **test/fixtures/**: YAML fixture files for water and CH3F
- **pyproject.toml**: Added `pyyaml` dependency

### Review fixes applied

- **Critical**: Geometry now required for all molecule entries — raises `ReferenceYAMLError` if missing (prevents molecule_idx mismatch causing `IndexError`)
- **High**: `data_idx` round-trip support — writer serializes `data_idx` when `atom_indices` is None; loader accepts either `atoms` or `data_idx`
- **Low**: Removed dead eigenmatrix code in `_parse_datum`
- 665 tests pass, lint clean

Closes #130

### CODE REVIEW

File: q2mm/parsers/reference_yaml.py [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+        ReferenceYAMLError: On malformed YAML or missing required fields.
+        FileNotFoundError: If *path* does not exist.
+
+    """
+    path = Path(path)
+    if not path.exists():
+        raise FileNotFoundError(f"Reference YAML file not found: {path}")
+
+    with open(path) as f:
+        raw = yaml.safe_load(f)
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/parsers/reference_yaml.py]: Fixed in bae2e62. Wrapped `yaml.safe_load()` in `try/except yaml.YAMLError` and re-raise as `ReferenceYAMLError`.

File: q2mm/parsers/reference_yaml.py [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+        return [
+            ReferenceValue(
+                kind="frequency",
+                value=value,
+                weight=weight,
+                label=label or f"mode {data_idx}",
+                molecule_idx=molecule_idx,
+                data_idx=data_idx,
+            )
+        ]
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/parsers/reference_yaml.py]: Fixed in dd858ff — same `data_idx >= 0` validation added for the single-frequency path. Test added.

File: q2mm/parsers/reference_yaml.py [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+    return kind
+
+
+def _as_float(value: Any, name: str, context: str) -> float:
+    """Coerce *value* to float with a clear error on failure."""
+    try:
+        return float(value)
+    except (TypeError, ValueError) as exc:
+        raise ReferenceYAMLError(f"'{name}' must be a number in {context}, got {value!r}.") from exc
+
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/parsers/reference_yaml.py]: Fixed in bcc7bf9. Added `isinstance(value, bool)` check to `_as_float()` matching the pattern in `_as_int()`. YAML `true`/`false` now raises `ReferenceYAMLError` instead of silently becoming 1.0/0.0. Added test `test_as_float_rejects_booleans`.

File: q2mm/parsers/reference_yaml.py [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+            atom_types=atom_types,
+            charge=charge,
+            multiplicity=multiplicity,
+            name=name,
+            bond_tolerance=bond_tolerance,
+        )
+
+    # ---- Hessian ----------------------------------------------------------
+    if "hessian" in mol_dict:
+        hessian_path = base_dir / mol_dict["hessian"]
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/parsers/reference_yaml.py]: Fixed in bcc7bf9. Same pattern as xyz — validates `hessian` is a string before path construction. Added test `test_hessian_non_string_rejected`.

File: q2mm/parsers/reference_yaml.py [hunk 5]
```diff
@@ -0,0 +1,581 @@
...
+        return [
+            ReferenceValue(
+                kind="eig_offdiagonal",
+                value=value,
+                weight=weight,
+                label=label or f"eig[{row},{col}]",
+                molecule_idx=molecule_idx,
+                atom_indices=(row, col),
+            )
+        ]
```
[the remark this answers is not in the document]
Comment (ericchansen) [reply q2mm/parsers/reference_yaml.py]: Fixed in bcc7bf9. Added `row < 0 or col < 0` guard with a descriptive error message. Added test `test_negative_eig_offdiagonal_row_col`.
